### PR TITLE
Add more info to output of `kit inspect`

### LINKED ISSUE #271: Improve inspect output
It should be possible to get additional information such as the Digest of the image. 

Other fields that can be considered are the cli version and creation date.

### PR DESCRIPTION
**Pull request by amisevsk**
### Description
Add additional information to the output of `kit inspect`:
```jsonc
{
  "digest": "<modelkit-digest>",
  "cliVersion": "<cli-version>",
  "kitfile": {
    // Kitfile
  },
  "manifest": {
    // Manifest  
  }
}
```

Note this change does mean that both `kit inspect` and `kit info` produce the Kitfile for the modelkit. I've left both commands intact because `kit info` outputs the Kitfile in YAML format (easier to read) whereas `inspect` outputs JSON. We can't yet produce yaml output for inspect since the OCI manifest struct does not have yaml tags and `gopkg.in/yaml.v3` does not respect json tags.

Note: to test version labelling, you will need to build `kit` with the version fields set:
```bash
go build -o kit -ldflags "-s -w -X kitops/pkg/cmd/version.Version=v0.0.1+test"
```

### Linked issues
Closes #271

### CODE REVIEW

File: pkg/cmd/inspect/inspect.go [hunk 1]
```diff
@@ -19,30 +19,53 @@ package inspect
...
 
-func getLocalManifest(ctx context.Context, opts *inspectOptions) (*ocispec.Manifest, error) {
+func getLocalManifest(ctx context.Context, opts *inspectOptions) (*inspectInfo, error) {
 	storageRoot := constants.StoragePath(opts.configHome)
 	store, err := repo.NewLocalStore(storageRoot, opts.modelRef)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read local storage: %w", err)
 	}
-	return repo.ResolveManifest(ctx, store, opts.modelRef.Reference)
+	desc, manifest, config, err := repo.ResolveManifestAndConfig(ctx, store, opts.modelRef.Reference)
```
Comment (gorkem) [inline pkg/cmd/inspect/inspect.go:52]: Can we replace `config` with `kitfile`? I always have to think for a few secs what is config in this context.

File: pkg/cmd/inspect/inspect.go [hunk 2]
```diff
@@ -19,30 +19,53 @@ package inspect
...
+	return &inspectInfo{
+		Digest:     desc.Digest,
+		CLIVersion: version,
+		Kitfile:    config,
+		Manifest:   manifest,
+	}, nil
 }
 
-func getRemoteManifest(ctx context.Context, opts *inspectOptions) (*ocispec.Manifest, error) {
+func getRemoteManifest(ctx context.Context, opts *inspectOptions) (*inspectInfo, error) {
```
Comment (gorkem) [inline pkg/cmd/inspect/inspect.go:68]: Is this repeating too much of the `getLocalManifest`? perhps we need to refactor this logic after repositroy creation to a function.

File: pkg/lib/repo/repo.go [hunk 3]
```diff
@@ -217,26 +217,30 @@ func GetConfig(ctx context.Context, store content.Storage, configDesc ocispec.De
 }
 
 // ResolveManifest returns the manifest for a reference (tag), if present in the target store
-func ResolveManifest(ctx context.Context, store oras.Target, reference string) (*ocispec.Manifest, error) {
+func ResolveManifest(ctx context.Context, store oras.Target, reference string) (ocispec.Descriptor, *ocispec.Manifest, error) {
 	desc, err := store.Resolve(ctx, reference)
 	if err != nil {
-		return nil, fmt.Errorf("reference %s not found in remote repository: %w", reference, err)
+		return ocispec.DescriptorEmptyJSON, nil, fmt.Errorf("reference %s not found in remote repository: %w", reference, err)
```
Comment (gorkem) [inline pkg/lib/repo/repo.go:223]: Should the error message change to remove "remote". I think this method can be called for local repo too.

**Review by gorkem — APPROVED** [review]
I have some suggestions but otherwise  :ok_hand:

File: pkg/cmd/inspect/inspect.go (hunk 1 above)
Comment (amisevsk) [reply pkg/cmd/inspect/inspect.go]: It all stems from the [original sin](https://github.com/jozu-ai/kitops/blob/bdceda24a2994947bbb2438103a895df278e4171/pkg/lib/storage/local.go#L61) :D

**Comment by amisevsk** [comment]
Applied fixes suggested above

**Pull request merged**